### ISSUE TITLE
Using $compile with a html fragment including text angular results in "An editor with name already exists..."

### ISSUE THREAD

**Issue by agairing** [fraywing/textAngular#795]
Hi,
I've got the following setup:
* div Element with text-angular and my custom directive
* my custom directive changes the div-Element and does a recompile
--> The recompile results in the error message "textAngular Error: An Editor with name "description" already exists ..."

See this plunker for details: "http://plnkr.co/edit/SX1K0rejsca0iDPg05p4" 

How can I prevent the error, but still do the recompile?

Additional question: 
The dynamic setting of the ng-model attribute works for the text field, but not for textangular. Any hint about this?

Any help is appreciated.

**Comment by pbassut** [comment]
That's because textAngular won't let you define two editors with the same name and when you recompile your directive angular applies textAngular directive all over again, thus resulting in the error.

You could call unregisterEditor before recompiling and then when the compiling is over textAngular will run as it is a brand new document.

**Comment by agairing** [comment]
Thanks for the answer. You mean "textAngularManager.unregisterEditor"? This doesn't help. 
Updated plunkr: http://plnkr.co/edit/0soxNjuB0YiUmViA5v10?p=preview

**Comment by nickknissen** [comment]
I am not completely sure what you are trying to achieve.
But if you want the `input` to contain a "clean" version of `model.html` without html tags you can do something like [this](http://plnkr.co/edit/fSFAM2rUuEsCB5LF4iNA?p=preview)

**Comment by pbassut** [comment]
Did @nickknissen solution worked for you? It did for me. Check if that's what you want so we can close this issue.

**Comment by agairing** [comment]
This doesn't solve my problem. My inital problem is that I want to set the "ng-model" directive via my own directive. In order to do this, I did a "recompile" of the html fragment containing the textangular-directive. If I do a recompile, the above error message is shown, because the textangular instance is registered a second time with the same name. Using "unregisterEditor" does not help - the error message still appears.

**Comment by nickknissen** [comment]
Have you tried not to use `name="description"` That seems to work for me.

**Comment by agairing** [comment]
Yes, this works. In that case a dynamic name is generated - and then the editor gets a different name in the recompile step. But the form validation is tricky without an expicit defined name. 
Maybe if I set the name also in my own directive, this could solve the problem....
Nevertheless, it would be nice if a html-fragment with a textangular directive and a name attribute can be recompiled.
